src/core/dep-graph.ts

```diff
@@ -331,12 +340,13 @@ class DepGraphImpl implements types.DepGraphInternal {
     for (const id of parentNodesIds) {
       if (ancestors.includes(id)) continue;
 
-      const pathToRoot = this.pathsFromNodeToRoot(id, ancestors).map((path) =>
-        [pkgInfo].concat(path),
-      );
+      const pathsFromNodeToRoot = this.pathsFromNodeToRoot(id, ancestors, opts);
 
-      for (const path of pathToRoot) {
-        allPaths.push(path);
+      pathsFromNodeToRoot.forEach((path) =>
```

Comment:
Yay, one loop instead of two. Not sure what purpose the `const` serves though. I'd way prefer a single `for of` here